numba_cuda/numba/cuda/tests/cudapy/test_caching.py

```diff
@@ -275,6 +224,37 @@ def f():
                 pass
 
 
+@skip_on_cudasim("Simulator does not implement caching")
+class CUDACooperativeGroupTest(SerialMixin, DispatcherCacheUsecasesTest):
```

Comment:
I think we need to preserve the comment about the original issue - I can imagine myself looking at this in future and not realising that we're specifically ensuring that caching works when function using CG is loaded from the cache prior to any other compilation:
```suggestion
class CUDACooperativeGroupTest(SerialMixin, DispatcherCacheUsecasesTest):
    # See Issue #9432: https://github.com/numba/numba/issues/9432
    # If a cached function using CG sync was the first thing to compile,
    # the compile would fail.
```